devel/api.md

```diff
@@ -0,0 +1,990 @@
+
+Example of changing eth1 MTU to 1460:
+
+```python
+import libnmstate
+from libnmstate.schema import Interface
+
+libnmstate.apply({
+    Interface.KEY: [
+        {
+            Interface.NAME: 'eth1',
+            Interface.MTU: 1460
+        }
+    ]
+})
+```
+
+## Manual Transaction Control
+
+## Error handling
+
+All the exceptions are stored in `libnmstate.error` and are all
+is inherited from `NmstateError`.
+
+### NmstateError
+
+The base exception of libnmstate.
+
+### NmstateDependencyError
+
+Nmstate requires external tools installed and/or started for desired state.
+
+### NmstateValueError
+
+Exception happens at pre-apply check, user should resubmit the amended
+desired state. Example:
+    * JSON/YAML syntax issue.
+    * Nmstate schema issue.
+    * Invalid value of desired property, like bond missing slave.
+
+### NmstatePermissionError
+
+Permission deny when applying the desired state.
+
+
+### NmstateConflictError
+
+Something else is already editing the network state via Nmstate.
+
+
+### NmstateLibnmError
+
+Exception for unexpected NetworkManager libnm failure.
+
+
+### NmstateVerificationError
+
+After applied desired state, current state does not match desired state for
+unknown reason.
+
+
+### NmstateNotImplementedError
+
+Desired feature is not supported by Nmstate yet.
+
+### NmstateInternalError
+
+Unexpected behaviour happened. It is a bug of libnmstate which should be fixed.
+
+## Schema
+
+The YAML schema file is placed at `libnmstate/schemas/operational-state.yaml`.
+
+In stead of using hard coded string, please use classes and constants in
+`libnmstate.schmea`.
+
+# Limitations
+
+* <https://nmstate.atlassian.net/browse/NMSTATE-235>
+
+  The `libnmstate` does not differentiate `InterfaceState.DOWN` and
+  `InterfaceState.ABSENT`. They both remove the on-disk Network Manager
+  connection profile of specified interface.
+
+* <https://nmstate.atlassian.net/browse/NMSTATE-236>
+
+  All the network state applied by `libnmstate` will be persistent after
+  reboot, except for disabling IPv6.
+
+* <https://nmstate.atlassian.net/browse/NMSTATE-115>
+
+  Due to the limitation of NetworkManager, Nmstate does not support disabling
+  DHCPv6 while enabling autoconf yet.
+
+* <https://nmstate.atlassian.net/browse/NMSTATE-212>
+
+  Currently, Nmstate requires the existence of interface section for all the
+  interfaces has route next hop to.
+
+# Network State Main layout
+
+The network state is represented in a dictionary:
+
+```python
+from libnmstate.schema import Interface
+from libnmstate.schema import Route
+from libnmstate.schema import DNS
+
+network_state = {
+    DNS.KEY: {},            # Check section of 'DNS'
+    Route.KEY: {},          # Check section of 'Route'
+    Interface.KEY: [],      # Check section of 'Interface -- Basic'
+}
+```
+
+# Basic Interface
+
+An interface represents a hardware or software network interface.
+Each interface will has these basic properties represented as a dictionary.
```

Comment:
Drop the `will`.
Consider removing the redundant "dictionary" text. This section can focus on the content.
(and it being a dict is obvious from the snippets)